### PR TITLE
Add automatic linting and formatting

### PR DESCRIPTION
**Pull request by aslakhellesoy**
We should add RuboCop and integrate it in the build in the same way as tslint/eslint for TypeScript.

This is a start, there are a lot of warnings to fix/ignore. The config and execution of rubocop also needs to be lifted up to `.templates/ruby`

### CODE REVIEW

**Comment by vincent-psarga** [comment]
I'm gonna hate myself for saying that, but why not [Standard](https://github.com/testdouble/standard) instead of managing our own rules for Rubocop ?

It might (will ...) be a real pain at first that said.

**Comment by aslakhellesoy** [comment]
I don't have an opinion on RuboCop vs Standard. We use RuboCop for cucumber-ruby. Any insights to offer @luke-hill @jaysonesmith @olleolleolle @mvz?

Anyone interested in completing this PR? I just wanted to kick-start it.

**Comment by vincent-psarga** [comment]
In fact Standard relies on Rubocop, the main idea is to comply to the ruby standard without having any choices in the rules (which somehow avoids discussions :D )

**Comment by luke-hill** [comment]
So I would look at this from 2 povs

1) maintenance. Asking community people to update rubocop is a (reasonably), understood task. Albeit in a repo like this quite big. But it's definitely do-able with little to no hand-holding required.
2) benefits vs others. I don't know standard, but if it just implements rubocop, and doesn't offer benefits, is it better to stick with rubocop? I would probably argue yes. It is one of the most maintained gems out there and is extremely reliable. The only issue is semver as its in the 0.x phase, but we can just be careful with our version

**Comment by luke-hill** [comment]
As for completion. I can happily pick this up in bits and pieces over the next few days. The fact you can have multiple todo files e.t.c. makes it quite easy to tackle as well.

**Comment by mvz** [comment]
I'm not thrilled by Standard's choices of how to configure RuboCop. The only benefit Standard has is that it offers no choice and would have everyone use the same style. However, since there is a whole history of different Ruby styles, that ship has sailed. Standard is certainly not 'the' ruby standard, but more the opinion of a small group of people.

TL;DR: I prefer plain RuboCop (plus rubocop-rspec), possibly with some light configuration and generous use of autogenerated TODO files.

**Comment by aslakhellesoy** [comment]
I started by adding RuboCop to Cucumber Expressions. As expected it found over 100 errors. Having the ability to tweak rules makes it easier to introduce lining without a huge up-front effort to fix all those errors. Given the history of this codebase I think a custom RuboCop config makes more sense. For a new codebase I'd happily go with Standard.
